internal/worker/worker.go

```diff
@@ -111,13 +128,21 @@ func (w *workerImpl) processJob(job *core.JobRecord, revision uint64) (bool, err
 	}
 
 	fmt.Printf("⚙️  Completed job %s, updating status to: %s\n", job.Id, job.Status)
-	_, finalUpdateErr := w.c.UpdateJob(job, newRevision)
-	if finalUpdateErr != nil {
-		// TODO: retry if error is transitional
-		return false, fmt.Errorf("Failed to update job %s: %v", job.Id, finalUpdateErr)
+	updateRetryTimer := time.NewTimer(finalUpdateRetryTimeout)
+finalUpdateLoop:
+	for {
+		select {
+		case <-updateRetryTimer.C:
+			return false, fmt.Errorf("Failed to update job %s within %d seconds", job.Id, finalUpdateRetryTimeout)
+		default:
+			_, finalUpdateErr := w.c.UpdateJob(job, newRevision)
+			if finalUpdateErr != nil {
+				log.Printf("Retrying final update to job %s: %v", job.Id, finalUpdateErr)
+				continue finalUpdateLoop
+			}
+			return false, nil
+		}
 	}
-
-	return false, nil
```

Comment:
I think it's clearer if the final return is out here, rather than 2 levels nested